packages/zksync-monitor/src/run.ts

```diff
@@ -0,0 +1,96 @@
+import assert from 'assert'
+import dotenv from 'dotenv'
+import { providers } from 'ethers'
+
+import {
+  Contracts,
+  GovernanceProxyCreationBlock,
+  Treasury,
+  Upgradeable,
+  UpgradeMasterOwners,
+  UpgradeMasterThreshold,
+  Validators,
+} from './constants'
+import { getMainnetSdk, MainnetSdk } from './eth-sdk/sdk'
+
+async function getUpgradeableContracts(
+  gateKeeper: MainnetSdk['upgradeGatekeeper'],
+) {
+  let shouldContinue = true
+  const managed: string[] = []
+  for (let i = 0; shouldContinue; ++i) {
+    try {
+      const address = await gateKeeper.managedContracts(i)
+      managed.push(address)
+    } catch {
+      shouldContinue = false
+    }
+  }
+  return managed
+}
+
+async function getActiveValidators(
+  governanceProxy: MainnetSdk['governanceProxy'],
+) {
+  const events = await governanceProxy.queryFilter(
+    governanceProxy.filters.ValidatorStatusUpdate(),
+    GovernanceProxyCreationBlock,
+  )
+  const activeValidators = events.reduce<string[]>((acc, { args: e }) => {
+    const idx = acc.indexOf(e.validatorAddress)
+    if (!e.isActive && idx !== -1) acc.splice(idx, 1)
+    if (e.isActive && idx === -1) acc.push(e.validatorAddress)
+    return acc
+  }, [])
+  return activeValidators
+}
+
+async function getAdditionalAddress(provider: providers.AlchemyProvider) {
+  const additionalAddressPosition = 19 // checked manually in contract code
+  const cellValue: string = await provider.getStorageAt(
+    Contracts.mainProxy,
+    additionalAddressPosition,
+  )
+  return '0x' + cellValue.slice(-40)
+}
+
+export async function run() {
+  dotenv.config()
+  const alchemyApiKey = process.env.ALCHEMY_API_KEY
+  const provider = new providers.AlchemyProvider('mainnet', alchemyApiKey)
```

Comment:
it's the very first time i have seen integration with RPC node done this way. 

If you use just a regular ethers JSON provider + put node URL in env you avoid alchemy lock-in.